recipes-soletta/dev-app/soletta-dev-app_0.0.7.bb

```diff
@@ -37,62 +33,57 @@ FILES_${PN} += " \
 
 SYSTEMD_SERVICE_${PN} = "soletta-dev-app-server.service soletta-dev-app-avahi-discover.service"
 
-do_compile() {
-
-    # changing the home directory to the working directory, the .npmrc will be created in this directory
-    export HOME=${WORKDIR}
-
-    # does not build dev packages
-    npm config set dev false
-
-    # access npm registry using http
-    npm set strict-ssl false
-    npm config set registry http://registry.npmjs.org/
-
-    # configure http proxy if neccessary
-    if [ -n "${http_proxy}" ]; then
-        npm config set proxy ${http_proxy}
-        export bower_https_proxy=${http_proxy}
-    fi
-    if [ -n "${HTTP_PROXY}" ]; then
-        npm config set proxy ${HTTP_PROXY}
-        export bower_https_proxy=${HTTP_PROXY}
-    fi
-
-    # configure cache to be in working directory
-    npm set cache ${WORKDIR}/npm_cache
-
-    # clear local cache prior to each compile
-    npm cache clear
-
-    case ${TARGET_ARCH} in
-        i?86) targetArch="ia32"
-            echo "targetArch = 32"
-            ;;
-        x86_64) targetArch="x64"
-            echo "targetArch = 64"
-            ;;
-        arm) targetArch="arm"
-            ;;
-        mips) targetArch="mips"
-            ;;
-        sparc) targetArch="sparc"
-            ;;
-        *) echo "unknown architecture"
-           exit 1
-            ;;
-    esac
-
-    npm install --arch=${targetArch} --production --verbose -g bower@v1.7.9
-
-    # compile and install node modules in source directory
-    npm --arch=${targetArch} --production --verbose install
-
-    bower -V install
+do_configure_append() {
+  export HOME=${WORKDIR}
+
+  npm config set dev false
+  npm set strict-ssl false
+  npm config set registry http://registry.npmjs.org/
+
+  if [ -n "${http_proxy}" ]; then
+      npm config set proxy ${http_proxy}
+  fi
+  if [ -n "${HTTP_PROXY}" ]; then
+      npm config set proxy ${HTTP_PROXY}
+  fi
+
+  case ${TARGET_ARCH} in
+      i?86) targetArch="ia32"
+          echo "targetArch = 32"
+          ;;
+      x86_64) targetArch="x64"
+          echo "targetArch = 64"
+          ;;
+      arm) targetArch="arm"
+          ;;
+      mips) targetArch="mips"
+          ;;
+      sparc) targetArch="sparc"
+          ;;
+      *) echo "unknown architecture"
+         exit 1
+          ;;
+  esac
+
+  #Installing bower a client-side libs manager
+  npm install --arch=${targetArch} --production --verbose -g bower@v1.7.9
+}
+
+do_compile_append() {
+
+  if [ -n "${http_proxy}" ]; then
+      export bower_https_proxy=${http_proxy}
+  fi
+  if [ -n "${HTTP_PROXY}" ]; then
+      export bower_https_proxy=${HTTP_PROXY}
+  fi
+
+  #Installing client-side libs
```

Comment:
@ipuustin, We've got a standalone package that contains all needed libs, and we can avoid fetching 24 libs from SRC_URI in this case. This quantity will grow in the future for sure.

I proposed this patch here: https://github.com/solettaproject/meta-soletta/pull/70

In my opinion, since, we've got to distribute these standalone package for clearlinux. It will be easier to us if we use it also in yocto.